### PR TITLE
add spec for String#split with capture and limit

### PR DESCRIPTION
**Pull request by CicholGricenchos**
related: https://github.com/opal/opal/issues/1599

### CODE REVIEW

File: core/string/split_spec.rb [hunk 1]
```diff
@@ -298,6 +298,8 @@
     "hi!".split(/()/, -1).should == ["h", "", "i", "", "!", "", ""]
     "hello".split(/((el))()/).should == ["h", "el", "el", "", "lo"]
     "AabB".split(/([a-z])+/).should == ["A", "b", "B"]
+    "aBBa".split(/(B)/, 2).should == ["a", "B", "Ba"]
+    "aBBa".split(/(B)/, 3).should == ["a", "B", "", "B", "a"]
```
Comment (eregon) [inline core/string/split_spec.rb]: Could you move these examples to a separate example (`it`) to emphase how they deal with the limit?

**Comment by eregon** [comment]
Thank you for the new examples.
I suppose this behavior is intentional, but it is indeed at least in contradiction with the documentation:
```
If limit is a positive number, at most that number of fields will be returned.
```

Therefore, could you report a bug on https://bugs.ruby-lang.org/ and ask for clarification if this is intended behavior and if so saying the documentation should be fixed (or even better, provide a patch)?

**Comment by CicholGricenchos** [comment]
Of course.
I have posted: https://bugs.ruby-lang.org/issues/13621 .

**Comment by eregon** [comment]
Seems a documentation issue indeed.

File: core/string/split_spec.rb (hunk 1 above)
Comment (eregon) [inline core/string/split_spec.rb]: Maybe:
```ruby
it "applies the limit only to the number of matches, not to the number of captures"
```

Comment (CicholGricenchos) [inline core/string/split_spec.rb]: @eregon Thanks for suggestion. The branch is updated.

**Comment by eregon** [comment]
Thank you for your contribution!

**Pull request merged**